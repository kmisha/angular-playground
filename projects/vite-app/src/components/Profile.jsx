export default function Profile({src, alt, id, isAvatar}) {
  return <img src={src} alt={alt} key={id} className={isAvatar && 'rounded-full'} />
}
