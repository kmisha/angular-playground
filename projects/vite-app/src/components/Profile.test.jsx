import renderer from 'react-test-renderer'
import {faker} from '@faker-js/faker'
import Profile from './Profile'

it('Passing props to the Profile component', () => {
    const component = renderer.create(
        <Profile src={faker.image.url} alt={faker.person.fullName} id={faker.number.int(100)} />
    )
    const tree = component.toJSON();

    expect(tree).toMatchSnapshot();
})

it('Using a rounded profile', () => {
    const component = renderer.create(
        <Profile src={faker.image.url} alt={faker.person.fullName} id={faker.number.int(100)} isAvatar={true} />
    )
    const tree = component.toJSON();

    expect(tree).toMatchSnapshot();
})
