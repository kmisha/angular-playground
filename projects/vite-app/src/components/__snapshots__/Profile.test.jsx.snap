// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Passing props to the Profile component 1`] = `
<img
  alt={[Function]}
  src={[Function]}
/>
`;

exports[`Using a rounded profile 1`] = `
<img
  alt={[Function]}
  className="rounded-full"
  src={[Function]}
/>
`;
